### PR TITLE
Placeholder string for Float::INFINITY in response

### PR DESCRIPTION
**Pull request by lpichler**
Chargeback tiers use`Float::INFINITY` in `finish` column but `Float::INFINITY` doesn't have representation in JSON. 
There is used "infinity" instead of `null`(before) because `null` just represents `null` values 

```
GET /api/rates/1850?expand=resources&attributes=chargeback_rate,detail_measure,detail_currency,chargeable_field,chargeback_tiers
```

```
{
    "name": "rates",
    "count": 138,
    "subcount": 138,
    "pages": 1,
    "resources": [
        {
            "href": "http://localhost:3090/api/rates/544",
            "id": "544",
            "enabled": true,
            "description": "Allocated CPU Count",
            "group": "cpu",
            "source": null,
            "metric": "derived_vm_numvcpus",
            "per_time": "hourly",
            "per_unit": "cpu",
            "friendly_rate": "Hourly @ 1.0 + 0.0 per Cpu from 0.0 to Infinity",
            "chargeback_rate_id": "8",
            "created_on": "2018-06-01T11:24:52Z",
            "updated_on": "2018-06-01T11:24:52Z",
            "chargeback_rate_detail_measure_id": null,
            "chargeback_rate_detail_currency_id": "1",
            "chargeable_field_id": "3",
            "sub_metric": null,
            "chargeback_tiers": [
                {
                    "id": "11010",
                    "chargeback_rate_detail_id": "544",
                    "start": 0.0,
                    "finish": "infinity",
                    "fixed_rate": 1.0,

```

See finish. 

@miq-bot assign @abellotti

### CODE REVIEW

**Comment by abellotti** [comment]
I see the change in the normalizer (rendering), what about as input to the API for creating rates, do we need to support the "infinity" string ?

**Comment by lpichler** [comment]
> I see the change in the normalizer (rendering), what about as input to the API for creating rates, do we need to support the "infinity" string ?

@abellotti yes exactly, we need and it was basically initiator of this fix -  it is included in this PR https://github.com/ManageIQ/manageiq-api/pull/847.

thanks

**Comment by Fryguy** [comment]
If we are handling infinity, should we also handle `Float:NAN` ?

**Comment by Fryguy** [comment]
Additionally, is it preferable to use `"Infinity"` vs `"infinity"`? I ask because infinity's representation in Javascript is with a capity `I`

```js
console.log(1 / 0)
// => Infinity
```

**Comment by lpichler** [comment]
> If we are handling infinity, should we also handle `Float:NAN` ?

I will do in other PR 👍 

and 

yes I agree, I changed `infinity`  to `Infinity`

thanks

File: spec/requests/chargebacks_spec.rb [hunk 1]
```diff
@@ -1,5 +1,31 @@
...
+    request_attributes = {:expand => 'resources', :attributes => 'chargeback_tiers,chargeback_rate,detail_measure,detail_currency,chargeable_field'}
+
+    get api_rates_url, :params => request_attributes
+
+    expect(response.parsed_body['resources'][0]['chargeback_tiers'][0]['id'].to_i).to eq(chargeback_tier.id)
+    expect(response.parsed_body['resources'][0]['chargeback_tiers'][0]['finish']).to eq('Infinity')
+    expect(response.parsed_body['resources'][0]['chargeback_rate']['id'].to_i).to eq(chargeback_rate_detail.chargeback_rate.id)
+    expect(response.parsed_body['resources'][0]['detail_measure']['id'].to_i).to eq(measure.id)
+    expect(response.parsed_body['resources'][0]['detail_currency']['id'].to_i).to eq(chargeback_rate_detail.detail_currency.id)
+    expect(response.parsed_body['resources'][0]['chargeable_field']['id'].to_i).to eq(chargeback_rate_detail.chargeable_field.id)
```
Comment (abellotti) [inline spec/requests/chargebacks_spec.rb:23]: I'm not crazy about the multitude of expects up there, can you replace with a single expect when you get a chance:

```
expect(response.parsed_body['resources'][0]).to include(
   "chargeback_tiers" => [
       {
          "id"     => chargeback_tier.id,
          "finish" => "Infinity"
       }
     ],
     "chargeback_rate" => {
          "id" => chargeback_rate_details.chargeback_rate.id
     ...
```


   

```

File: spec/requests/chargebacks_spec.rb [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+
+    expect(response.parsed_body)
```
Comment (abellotti) [inline spec/requests/chargebacks_spec.rb:25]: Is this needed ?

Comment (lpichler) [reply spec/requests/chargebacks_spec.rb:25]: Removed 👍

File: spec/requests/chargebacks_spec.rb (hunk 1 above)
Comment (lpichler) [reply spec/requests/chargebacks_spec.rb]: I used `include` now 👍

**Comment by abellotti** [comment]
Thanks @lpichler for fixing this. 👍

**Comment by simaishi** [comment]
Jansa backport details:

```
$ git log -1
commit 80cbbfba8dafb6252e5c4d263245717d2ade0735
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Tue Jun 2 09:40:48 2020 -0400

    Merge pull request #851 from lpichler/display_infinity

    Placeholder string for Float::INFINITY in response

    (cherry picked from commit 4601a36691fc60033577092d295b909b54588289)
```

**Pull request merged**